test/integration-tests/Seaport.test.ts

```diff
@@ -141,4 +157,187 @@ describe('Seaport', () => {
+      planner = new RoutePlanner()
+      tubbyCats = new ethers.Contract(TUBBY_ADDRESS, ERC721_ABI) as ERC721
+      weth = new ethers.Contract(WETH.address, ERC20_ABI) as ERC20
+
+      // send 1 eth from alice to router.address for weth approval
+      await alice.sendTransaction({ to: router.address, value: ethers.utils.parseEther('1.0') })
+      expect(await ethers.provider.getBalance(router.address)).to.eq(ethers.utils.parseEther('1.0'))
+      // max approve conduit for weth
+      const routerSigner = await ethers.getImpersonatedSigner(router.address)
+      await weth.connect(routerSigner).approve(OPENSEA_CONDUIT, ethers.constants.MaxUint256)
+      await weth.connect(routerSigner).allowance(router.address, OPENSEA_CONDUIT)
+      expect(await weth.connect(routerSigner).balanceOf(router.address)).to.eq(0)
+    })
+
+    it('accepts an advanced order with criteria offering WETH', async () => {
+      // https://etherscan.io/tx/0x74551f604adea1c456395a8e801bb063bbec385bdebbc025a75e0605910f493c
+      let { advancedOrder, criteriaResolvers } = getAdvancedOrderParams(seaportOrders[2])
+      const value = calculateValue(advancedOrder.parameters.consideration, [ItemType.ERC20])
+      const params = advancedOrder.parameters
+      // Can add logic to select approval target (conduit or consideration) depending on conduitHash
+      // TODO: add helper function to get identifier from criteriaResolvers if exists
+      const id = criteriaResolvers[0].identifier
+      // transfer nft to alice as tubbyCatOwner
+      const prevTubbyCatOwner = await tubbyCats.connect(alice).ownerOf(id)
+      await tubbyCats
+        .connect(await ethers.getImpersonatedSigner(prevTubbyCatOwner))
+        .transferFrom(prevTubbyCatOwner, alice.address, id)
+
+      const wethReceived = BigNumber.from(advancedOrder.parameters.offer[0].startAmount).sub(value)
+
+      tubbyCats = tubbyCats.connect(alice)
+      const ownerBefore = await tubbyCats.ownerOf(id)
+      expect(ownerBefore).to.eq(alice.address)
+
+      const calldata = seaportInterface.encodeFunctionData('fulfillAdvancedOrder', [
+        advancedOrder,
+        criteriaResolvers,
+        OPENSEA_CONDUIT_KEY,
+        ADDRESS_ZERO, // 0 addr so router custody
+      ])
+
+      // TODO: need to add arg for msg.value?
+      planner.addCommand(CommandType.SEAPORT_SELL_721, [
+        calldata,
+        tubbyCats.address,
+        OPENSEA_CONDUIT,
+        id,
+        alice.address,
+      ])
+      planner.addCommand(CommandType.SWEEP, [WETH.address, MSG_SENDER, 0])
+      const { commands, inputs } = planner
+
+      const wethBefore = await weth.connect(alice).balanceOf(alice.address)
+      expect(wethBefore).to.eq(0)
+      // put NFT in the router TODO replace with Permit2 721
+      await tubbyCats.transferFrom(alice.address, router.address, id)
+      expect(await tubbyCats.ownerOf(id)).to.eq(router.address)
```

Comment:
also don't feel this is necessary, line 229 should just throw an errror if something goes wrong

Reply:
true, I may have gotten carried away w using expects for debugging